recipe/build.sh

```diff
@@ -19,31 +19,52 @@ sed -i.bak "s/@DALI_INSTALL_REQUIRES_NVIMGCODEC@//g" dali/python/setup.py.in
 mkdir -p build
 cd build
 
+DALI_LINKING_ARGS=(
+  -DLINK_DRIVER=OFF
+# Continue to dlopen nvimgcodec so that it can be optionally installed
+  -DWITH_DYNAMIC_NVIMGCODEC=ON
+  -DNVIMGCODEC_DEFAULT_INSTALL_PATH=${PREFIX}
+# Disable all dynamic (dlopen) linkages because we have patched the static links away
+  -DWITH_DYNAMIC_CUDA_TOOLKIT=OFF
+  -DWITH_DYNAMIC_CUFFT=OFF
+  -DWITH_DYNAMIC_NPP=OFF
+  -DWITH_DYNAMIC_NVJPEG=OFF
```

Comment:
The preference is to link dynamically to shrink the size of the binary.

Reply:
As noted in the comment. We have patched the static links away. These libraries are linked dynamically, just not with dlopen.

Reply:
Any particular reason to move away from dlopen in favor of this approach which diverges from the upstream code?

Reply:
Lets us check at build time if runtime dependencies are satisfied by inspecting the binary elf record for links and undefined symbols. The build runners don't have GPUs so we can't actually load the binaries to see if they fail because trying to load the cuda driver will always fail.

What's the reason for using dlopen?

Reply:
Making CUDA optional at runtime with a single build variant is a strong reason to use dlopen.

Reply:
I'm very biased so a second opinion is invaluable for me.